lms/djangoapps/courseware/tests/test_group_access.py

```diff
@@ -116,10 +125,15 @@ def setUp(self):
         self.course = CourseFactory.create(
             user_partitions=[self.animal_partition, self.color_partition],
         )
-        self.chapter = ItemFactory.create(category='chapter', parent=self.course)
-        self.section = ItemFactory.create(category='sequential', parent=self.chapter)
-        self.vertical = ItemFactory.create(category='vertical', parent=self.section)
-        self.component = ItemFactory.create(category='problem', parent=self.vertical)
+        chapter = ItemFactory.create(category='chapter', parent=self.course)
```

Comment:
Using the stored xblocks causes issues as they may not reflect updates (with split). So just store the locations and look up the xblocks when needed.